### PR TITLE
Fixes for PHP 7.3

### PR DESCRIPTION
**Pull request by michael-e**
Re: #2884

### CODE REVIEW

File: symphony/lib/toolkit/class.json.php [hunk 1]
```diff
@@ -5,13 +5,13 @@
...
 /**
- * The `JSONException` class extends the base `Exception` class. It's only
+ * The `SymJSONException` class extends the base `Exception` class. It's only
  * difference is that it will translate the `$code` to a human readable
  * error.
  *
  * @since Symphony 2.3
  */
-class JSONException extends Exception
+class SymJSONException extends Exception
```
Comment (brendo) [inline symphony/lib/toolkit/class.json.php]: I'm curious, would a better fix be to only create this exception if it didn't exist? What sort of error messages does the new native `JsonException` give? Are they are descriptive as ours?

Comment (michael-e) [reply symphony/lib/toolkit/class.json.php]: That was [sugested](https://github.com/symphonycms/symphony-2/issues/2884#issuecomment-466185413) by @nitriques as well... Honestly I haven't tested anything here — I simply needed to make it run on PHP 7.3. Unfortunatley, the PHP docs don't telll me much about how descriptive the exception will be. But if it is desired, I can change the code acoordingly.

Comment (nitriques) [reply symphony/lib/toolkit/class.json.php]: The code for the native JSONException is almost identical to our implementation: https://github.com/php/php-src/blob/623911f993f39ebbe75abe2771fc89faf6b15b9b/ext/json/json.c#L215

I would not mind dropping our version. We may loose the translation, but will always support all error codes. I do not mind loosing the translation (it is not meant to be user facing anyways).

I'll fix it with a class exists check.

File: symphony/lib/toolkit/class.general.php [hunk 2]
```diff
@@ -344,7 +344,7 @@ public static function createHandle($string, $max_length = 255, $delim = '-', $u
         $string = preg_replace('/[\s]+/', $delim, $string);
 
         // Find all legal characters
-        preg_match_all('/[^<>?@:!-\/\[-`;‘’…]+/u', $string, $matches);
+        preg_match_all('/[^<>?@:!\-\/\[-`;‘’…]+/u', $string, $matches);
```
Comment (nitriques) [inline symphony/lib/toolkit/class.general.php]: This is backward compatible right ?

File: symphony/lib/toolkit/class.json.php (hunk 1 above)
Comment (michael-e) [reply symphony/lib/toolkit/class.json.php]: Please go ahead! Thanks!

File: vendor/composer/ClassLoader.php [hunk 3]
```diff
@@ -279,7 +279,7 @@ public function isClassMapAuthoritative()
      */
     public function setApcuPrefix($apcuPrefix)
     {
-        $this->apcuPrefix = function_exists('apcu_fetch') && ini_get('apc.enabled') ? $apcuPrefix : null;
+        $this->apcuPrefix = function_exists('apcu_fetch') && filter_var(ini_get('apc.enabled'), FILTER_VALIDATE_BOOLEAN) ? $apcuPrefix : null;
```
Comment (nitriques) [inline vendor/composer/ClassLoader.php]: Is this auto-generated ? Or custom fixes ?

File: symphony/lib/toolkit/class.general.php (hunk 2 above)
Comment (michael-e) [reply symphony/lib/toolkit/class.general.php]: Yes, definitely.

File: vendor/composer/ClassLoader.php (hunk 3 above)
Comment (michael-e) [reply vendor/composer/ClassLoader.php]: Yes, that was generated by Composer.

Comment (michael-e) [reply vendor/composer/ClassLoader.php]: It was:

> Composer version 1.8.0 2018-12-03 10:31:16

File: symphony/lib/toolkit/class.json.php (hunk 1 above)
Comment (michael-e) [reply symphony/lib/toolkit/class.json.php]: Oh, I misunderstood. Should I add a commit to my PR?

**Comment by michael-e** [comment]
I think that we don't need to run `composer dump-autoload` anymore, because `autoload_classmap.php` wouldn't change. (FYI, if I try, I get the same programmatic differences in `ClassLoader.php` as before.)

File: symphony/lib/toolkit/class.json.php (hunk 1 above)
Comment (michael-e) [reply symphony/lib/toolkit/class.json.php]: Done.

**Comment by michael-e** [comment]
@nitriques: I added a commit with the requirements change. This might cause some headache for extension developers. Unfortunately, we never officially added the PHP compatibility attributes to the extension meta XML schema. 

However, my big system (on PHP 7.3) uses a lot of extensions, and I have only found a few issues (and sent PRs). And the more important point is: Symphony LTS should officially run on the current PHP version. I'd rather risk a bug than make Symphony look "discontinued".

**Comment by nitriques** [comment]
> Unfortunately, we never officially added the PHP compatibility attributes to the extension meta XML schema.

We did it ;) it is not released yet...

> Symphony LTS should officially run on the current PHP version. 

Yes, but we can't drop anything!

**Comment by froschdesign** [comment]
@nitriques 
> Yes, but we can't drop anything!

Sure you can! Or does someone pay you to support the dead PHP versions?

And it is a wrong signal to the users of the CMS, because if you support the old PHP versions the user believes he does not have to do anything. But it is a security risk!

As a (open source) project without charge, it should not be responsible for problems with unsupported PHP versions.

**Comment by froschdesign** [comment]
Look also at semantic versioning: "[What should I do if I update my own dependencies without changing the public API?](https://semver.org/#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api)"

**Pull request merged**